dot/rpc/modules/author_integration_test.go

```diff
@@ -395,6 +385,108 @@ func TestAuthorModule_HasKey_Integration(t *testing.T) {
 	}
 }
 
+func TestAuthorModule_HasSessionKeys_Integration(t *testing.T) {
+	tmpdir := t.TempDir()
+	intCtrl := setupStateAndRuntime(t, tmpdir)
+	intCtrl.stateSrv.Transaction = state.NewTransactionState()
+	intCtrl.keystore = keystore.NewGlobalKeystore()
+
+	auth := setupAuhtorModule2Test(t, intCtrl)
+
+	const granSeed = "0xf25586ceb64a043d887631fa08c2ed790ef7ae3c7f28de5172005f8b9469e529"
+	const granPubK = "0x6b802349d948444d41397da09ec597fbd8ae8fdd3dfa153b2bb2bddcf020457c"
+
+	const sr25519Seed = "0xe5be9a5092b81bca64be81d212e7f2f9eba183bb7a90954f7b76361f6edb5c0a"
+	const sr25519Pubk = "0xd43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d"
+
+	insertSessionKeys := []struct {
+		ktype      []string
+		seed, pubk string
+	}{
+		{
+			ktype: []string{"gran"},
+			seed:  granSeed,
+			pubk:  granPubK,
+		},
+		{
+			ktype: []string{"babe", "imon", "audi"},
+			seed:  sr25519Seed,
+			pubk:  sr25519Pubk,
+		},
+	}
+
+	for _, toInsert := range insertSessionKeys {
+		for _, keytype := range toInsert.ktype {
+			err := auth.InsertKey(nil, &KeyInsertRequest{
+				Type:      keytype,
+				Seed:      toInsert.seed,
+				PublicKey: toInsert.pubk,
+			}, nil)
+			require.NoError(t, err)
+		}
+	}
+
+	testcases := map[string]struct {
+		pubSessionKeys string
+		expect         bool
+		waitErr        error
+	}{
+		"public keys are in the right order, should return true": {
```

Comment:
nit Don't change here, just for future tests - I'm not sure how punctuation is handled for subtests, they might be removed or changed to `_`, so in the future try to have test case names without punctuation i.e. `"public keys are in the right order returns true"`. No need to say the return either I would say since it's in the test case values, but that's just my opinion :wink:

Reply:
Got it!